### PR TITLE
bump k8s deps to v1.28.4

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

With completion of #662, we should bump k8s deps in `master` to v1.28.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
bump k8s deps to v1.28.4
```

### CODE REVIEW

**Comment by Huang-Wei** [comment]
The CI failure is valid. It hits an issue/regression on [RunScorePlugins](https://github.com/kubernetes/kubernetes/blob/v1.28.4/pkg/scheduler/framework/runtime/framework.go#L1019) of upstream:

```go
numPlugins := len(f.scorePlugins) - state.SkipScorePlugins.Len()
```

If a plugin implements both PreScore and Score, numPlugins' could be negative if:

- **only** Score is disabled
- `Skip` is returned in PreScore

**Comment by Huang-Wei** [comment]
cc @denkensk

**Comment by Huang-Wei** [comment]
(@denkensk was in a business trip)

@ffromani @zwpaper could you help review this PR? we should get it to master as early as we can. 🙇‍♂️

File: cmd/scheduler/main_test.go [hunk 1]
```diff
@@ -497,18 +497,18 @@ profiles:
...
+			name:            "single profile config - PodState - v1beta3",
 			flags:           []string{"--config", podStateConfigFile},
 			registryOptions: []app.Option{app.WithPlugin(podstate.Name, podstate.New)},
 			wantPlugins: map[string]*config.Plugins{
 				"default-scheduler": {
-					PreEnqueue: defaults.PluginsV1beta2.PreEnqueue,
-					QueueSort:  defaults.PluginsV1beta2.QueueSort,
-					Bind:       defaults.PluginsV1beta2.Bind,
-					PostFilter: defaults.PluginsV1beta2.PostFilter,
+					PreEnqueue: defaults.ExpandedPluginsV1beta3.PreEnqueue,
```
Comment (ffromani) [inline cmd/scheduler/main_test.go:505]: why are we moving from `Plugins*` to `ExpandedPlugins*`? I see other occurrences in the old code, but still curious

Comment (Huang-Wei) [reply cmd/scheduler/main_test.go:505]: v1beta2 was removed in 1.28, so we have to turn to v1beta3. While in v1beta3, the variable `defaults.PluginsV1beta3` refers to a special plugin list with `MultiPoint` extension point:

```go
var PluginsV1beta3 = &config.Plugins{
	MultiPoint: config.PluginSet{...}
}
```

 so we have to use `ExpandedPluginsV1beta3` instead.

File: test/util/fake.go [hunk 2]
```diff
@@ -146,45 +167,54 @@ func (n *nominator) add(pi *framework.PodInfo, nominatingInfo *framework.Nominat
...
 	}
-	delete(n.nominatedPodToNode, p.UID)
+	delete(npm.nominatedPodToNode, p.UID)
 }
 
 // UpdateNominatedPod updates the <oldPod> with <newPod>.
-func (n *nominator) UpdateNominatedPod(oldPod *v1.Pod, newPodInfo *framework.PodInfo) {
-	n.Lock()
-	defer n.Unlock()
+func (npm *nominator) UpdateNominatedPod(logger klog.Logger, oldPod *v1.Pod, newPodInfo *framework.PodInfo) {
```
Comment (denkensk) [inline test/util/fake.go:211]: Do we need to replace 'n' with 'npm'? I recall that 'npm' is an abbreviation for 'nominatePodMap '. or just copy from upstream？

Comment (Huang-Wei) [reply test/util/fake.go:211]: I didn't notice that - just copied from upstream.

**Comment by Huang-Wei** [comment]
Thanks @ffromani and @denkensk .

/unhold

File: cmd/scheduler/main_test.go (hunk 1 above)
Comment (ffromani) [reply cmd/scheduler/main_test.go:505]: thanks for the clarification

**Pull request merged**